.github/workflows/nodejs.yml

```
@@ -36,5 +41,17 @@ jobs:
       run: npm i
 
     - name: Test
-      run: npm test
+      id: test
+      run: |
+        npm run test-coverage-ci || echo "Silently ignoring coverage threshold limit..."
+
```

```suggestion
      run: npm test

    - name: Test Coverage
      id: test
      run: |
        npm run test-coverage-ci || echo "Silently ignoring coverage threshold limit..."

```

I humbly would like to suggest this change, until we close #240 and revert 222a3dcb523b0b5542d387c4b503f515ac60a9e8. For me, this seemed to be a little outside of the scope, hence thought of asking first.

This will restrict from pushing commits that break existing tests, otherwise everything will be ignored, and it would take manual effort to check whether tests are failing.

---

@CyberCitizen01 - I'm a bit confused, why do we need to run the tests twice?

---

The following is the current behaviour:
- `npm run test-coverage-ci` (analyse coverage and generate `lcov.info`) -> `npm test` -> `mocha --exit` (run tests and exit with non-zero code if any test fails)
- The expression
  ```bash
  npm run test-coverage-ci || echo "Silently ignoring coverage threshold limit..."
  ```
  basically does, "Run, `npm run test-coverage-ci` and if it fails don't propagate the error code, instead run `echo "Silently ... "` and exit with `0` code

The problem, with the above approach, that fears me is that suppose a PR has modified something that breaks the existing code, it'll go unnoticed even though `npm test` exits with non-zero code, because it is essentially being ignored because of the `||`.

For now, the best suggestion I have is to add an additional step to specifically run tests (`npm test`). We'll only need to do this until the aforementioned conditions are met.

---

Running the tests twice is just a quick and temporary fix, until we reach 80% coverage (#240). After that, we'll just revert the suggestions made in https://github.com/finos/git-proxy/pull/213#discussion_r1367778916 thread.

And, too explain further:
Tests can fail, irrespective of the code coverage report generation, it's just that both failing tests and code coverage error out, and since we have removed explicit failure (as suggested earlier), we ignore both aspects of failing tests and code coverage. Running the tests again will make sure that, tests are passing and nothing breaks.

There's one more way to do it without having to run the tests twice - we can try to parse the output of `npm run test-coverage-ci` and try to make sense of it. But this will get messy, and to be honest may not be reliable.